WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/packages/WooShippingLabelPackageCreationViewModel.kt

```diff
@@ -297,7 +306,7 @@ class WooShippingLabelPackageCreationViewModel @Inject constructor(
             ?.model?.firstOrNull()
             ?.let { PackageData.fromPackageDAO(it) }
             ?.let { Result.success(it) }
-            ?: Result.failure(Throwable("Failed to save package"))
+            ?: Result.failure(Throwable(response.error.type.toString()))
```

Comment:
I think using `WooException` here would be better as it exposes both the error type and message.

```suggestion
            ?: Result.failure(WooException(response.error))
```

Reply:
For now, I only need the error type for tracking. The original parameter for `Result.failure` is `Throwable`. If I use `WooException` here, I would need to cast it to `WooException` to access the type. Additionally, the consumer function would need to be aware that `sendCustomPackageToStore()` returns a `WooException`. I think using `Throwable` is a simpler approach in this case.

Reply:
> Additionally, the consumer function would need to be aware that sendCustomPackageToStore() returns a WooException

Why? We use `fold` in the consumer function, and and this will continue to work, `WooException` is a `Throwable`.

> For now, I only need the error type for tracking

This is related to my other [question](https://github.com/woocommerce/woocommerce-android/pull/14323#discussion_r2207740096), it seems to me we report the [`message`](https://github.com/woocommerce/woocommerce-android/blob/4216b383c1ec1197d4345e8694690b88fdee022d/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationViewModel.kt#L600) in some places, and the `type` in other places.

Reply:
> Why? We use fold in the consumer function, and and this will continue to work, WooException is a Throwable.

Yes, but `Throwable` doesn't have a `type` property. To access the `type` from `WooException`, I'd need to cast `Throwable` to `WooException`. That said, your suggestion will eventually be necessary when we start including more error data in events. So, I'll leave using `WooException` for PR where we add `error_context`, `error_description` and `error_type`.